src/mixins/responsiveHandler.js

```diff
@@ -1,57 +1,68 @@
 export const responsiveHandler = {
-    data() {
-        return {
-            element: null,
-            elementWidth: 0,
-            elementHeight: 0,
-            deviceSize: 4,
-            deviceSizes: {
-                s: 1,
-                m: 2,
-                l: 3,
-                xl: 4
-            }
-        }
+  data() {
+    return {
+      element: null,
+      elementWidth: 0,
+      elementHeight: 0,
+      deviceSize: 4,
+      deviceSizes: {
+        s: 1,
+        m: 2,
+        l: 3,
+        xl: 4
+      }
+    }
+  },
+  computed: {
+    isSmallOrDown() {
+      return this.deviceSize == this.deviceSizes.s
+    },
+    isMediumOrDown() {
+      return this.deviceSize <= this.deviceSizes.m
     },
-    mounted() {
-        this.$nextTick(() => {
-            this.element = window
-            if (this.element) {
-                window.addEventListener('resize', this.setElementWidth)
-                window.addEventListener('resize', this.setElementHeight)
-                this.setElementWidth()
-                this.setElementHeight()
-            }
-        })
+    isLargeOrDown() {
+      return this.deviceSize <= this.deviceSizes.l
+    }
+  },
+  mounted() {
+    this.$nextTick(() => {
+      this.element = window
+      if (this.element) {
+        window.addEventListener('resize', this.setElementWidth)
+        window.addEventListener('resize', this.setElementHeight)
+        this.setElementWidth()
+        this.setElementHeight()
+      }
+    })
+  },
+  methods: {
+    setElementWidth(event) {
+      this.elementWidth = this.element.innerWidth
+      this.setDeviceSize(this.elementWidth)
     },
-    methods: {
-        setElementWidth(event) {
-            this.elementWidth = this.element.innerWidth
-            this.setDeviceSize(this.elementWidth)
-        },
-        setElementHeight(event) {
-            this.elementHeight = this.element.innerHeight
-        },
-        setDeviceSize(width) {
-            if (width > 1200) {
-                // large desktop
-                this.deviceSize = 4
-            } else if (width > 991 && width <= 1200) {
-                // laptop
-                this.deviceSize = 3
-            } else if (width > 600 && width <= 991) {
-                // tablet
-                this.deviceSize = 2
-            } else if (width <= 600) {
-                // smartphone
-                this.deviceSize = 1
-            }
-        }
+    setElementHeight(event) {
+        this.elementHeight = this.element.innerHeight
     },
-    beforeDestroy() {
-        if (this.element) {
-            window.removeEventListener('resize', this.setElementWidth)
-            window.removeEventListener('resize', this.setElementHeight)
-        }
+    setDeviceSize(width) {
+      if (width > 1200) {
+        // large desktop
+        this.deviceSize = 4
```

Comment:
@KevinTss maybe best to reuse the objects `deviceSizes.xl` instead of setting the value `4` directly?

Reply:
This one is for use the existing `responsiveHandler`. My opinion is that we can refactor it in a separate PR
Is a very good remark